hack/generator/pkg/codegen/pipeline_create_arm_types.go

```diff
@@ -48,15 +48,17 @@ func createArmTypesAndCleanKubernetesTypes(idFactory astmodel.IdentifierFactory)
 	}
 }
 
-func createArmTypes(definitions astmodel.Types) (astmodel.Types, astmodel.Types, error) {
+func createArmTypes(
+	definitions astmodel.Types,
+	idFactory astmodel.IdentifierFactory) (astmodel.Types, astmodel.Types, error) {
```

Comment:
it might be nice to definer a receiver type with `definitions` & `idFactory` on it.

Reply:
Agree - did this and reorganized this file -- will send it in a separate PR once I coordinate with @theunrepentantgeek